Code/Framework/AzCore/Platform/Linux/AzCore/std/parallel/internal/thread_Linux.cpp

```diff
@@ -53,7 +54,15 @@ namespace AZStd
 
         void PostCreateThread(pthread_t tId, const char* name, int)
         {
-            pthread_setname_np(tId, name);
+            if (const int result = pthread_setname_np(tId, name); result == ERANGE)
+            {
+                // The name was too long. pthread limits the name to 16
+                // characters (including the null terminator). Let's elide the
+                // middle.
+                const auto len = strlen(name);
+                const auto elidedName = AZStd::fixed_string<16>{name, 7} + "." + AZStd::fixed_string<16>{name + len - 7, 7};
+                pthread_setname_np(tId, elidedName.c_str());
```

Comment:
I think you should check the results of the second attempt as well, in case of invalid characters or something? (I don't know what the order of error checking ```pthread_setname_np``` will do)

Reply:
According to the pthread documentation, the only way that `pthread_setname_np` will error is with `ERANGE`:

```
ERRORS
       The pthread_setname_np() function can fail with the following error:

       ERANGE The length of the string specified pointed to by name exceeds the allowed limit.
```

Reply:
Wonderful